test/GsnCapableDocumentStore.js

```diff
@@ -0,0 +1,400 @@
+    const documentHash = "0x10327d7f904ee3ee0e69d592937be37a33692a78550bd100d635cdea2344e6c7";
+
+    it("returns false for document issued after the block number", async () => {
+      const issueReceipt = await instance.issue(documentHash);
+      const issuedBlock = get(issueReceipt, "receipt.blockNumber");
+      const issued = await instance.isIssuedBefore(documentHash, issuedBlock - 1);
+      expect(issued).to.be.false;
+    });
+
+    it("returns true for document issued at the block number", async () => {
+      const issueReceipt = await instance.issue(documentHash);
+      const issuedBlock = get(issueReceipt, "receipt.blockNumber");
+      const issued = await instance.isIssuedBefore(documentHash, issuedBlock);
+      expect(issued).to.be.true;
+    });
+
+    it("returns true for document issued before the block number", async () => {
+      const issueReceipt = await instance.issue(documentHash);
+      const issuedBlock = get(issueReceipt, "receipt.blockNumber");
+      const issued = await instance.isIssuedBefore(documentHash, issuedBlock + 1);
+      expect(issued).to.be.true;
+    });
+
+    it("returns false for document not issued, for arbitary block number", async () => {
+      const issued = await instance.isIssuedBefore(documentHash, 1000);
+      expect(issued).to.be.false;
+    });
+
+    it("returns false for document not issued, for block 0", async () => {
+      const issued = await instance.isIssuedBefore(documentHash, 0);
+      expect(issued).to.be.false;
+    });
+  });
+
+  describe("trustedForwarder", () => {
+    it("returns trustedForwarder address", async () => {
+      const trustedForwarder = await instance.getTrustedForwarder();
+      expect(trustedForwarder).to.be.equal(accounts[1]);
+    });
+
+    it("should set trustedForwarder to given address", async () => {
+      await instance.setTrustedForwarder(accounts[2]);
+      const trustedForwarder = await instance.getTrustedForwarder();
+      expect(trustedForwarder).to.be.equal(accounts[2]);
+    });
+  });
+
+  describe("supportsInterface", () => {
+    it("returns true if supportsInterface for GsnCapable", async () => {
+      const supportsGsnCapableInterface = await instance.supportsInterface("0xa5a23640");
+      expect(supportsGsnCapableInterface).to.be.true;
+    });
+
+    it("return false if does not support given interface", async () => {
+      const supportsGsnCapableInterface = await instance.supportsInterface("0xffffffff");
+      expect(supportsGsnCapableInterface).to.be.false;
+    });
+  });
+
+  describe("paymaster", () => {
+    it("should set paymaster address", async () => {
+      const receipt = await instance.setPaymaster(accounts[2]);
+
+      expect(receipt.logs[0].event).to.be.equal("PaymasterSet");
+      expect(receipt.logs[0].args.target).to.be.equal(accounts[2]);
+
+      const paymaster = await instance.getPaymaster();
+      expect(paymaster).to.be.equal(accounts[2]);
+    });
+  });
```

Comment:
no test to show that someone else can actually submit the transaction on behalf of the owner?